src/main/webapp/app/exercises/shared/exercise-scores/exercise-scores.component.ts

```diff
@@ -382,20 +387,20 @@ export class ExerciseScoresComponent implements OnInit, OnDestroy {
         }
     }
 
-    getAssessmentLink(participation: Participation) {
-        if (!this.exercise.type || !this.course.id || !this.exercise.id || !participation.results?.[0]?.submission?.id) {
+    getAssessmentLink(participation: Participation, correctionRound = 0) {
+        const newAssessment = !participation.results?.[correctionRound]?.assessmentType || participation.results?.[correctionRound].assessmentType === AssessmentType.AUTOMATIC;
+        if (!this.exercise.type || !this.exercise.id || !this.course.id || (!participation.results?.[correctionRound]?.submission?.id && !newAssessment)) {
             return;
         }
-        const newAssessment = !participation.results?.[0]?.assessmentType || participation.results?.[0].assessmentType === AssessmentType.AUTOMATIC;
         return getLinkToSubmissionAssessment(
             this.exercise.type,
             this.course.id,
             this.exercise.id,
             participation.id,
-            newAssessment ? 'new' : participation.results[0].submission.id,
+            newAssessment ? 'new' : participation.results![correctionRound].submission!.id!,
             this.exercise.exerciseGroup?.exam?.id,
             this.exercise.exerciseGroup?.id,
-            participation.results[0].id,
+            participation.results?.[correctionRound]?.id,
```

Comment:
Why do you assume the results as optional here, but also required in line 400?

Reply:
In line 400 is the ternary that makes sure that the results are present

Reply:
I'm not talking about the teneary itself.

line 403: `participation.results?.[correctionRound]?.id`
line 400: `participation.results![correctionRound].submission!.id!`

One time `results?.`, one time `results!`. Either you are sure about the results array existing, then you should mark it in line 403 as existing too, or you aren't, then the results array in line 400 could be undefined too.

Reply:
The condition in the ternary is making sure that the results are present in the else part of the ternary.
So I'm sure that the results array is present in line 400 but not in 403

Reply:
Ah, I get it. I rechhecked where newAssessment comes from. Makes sense then. Sorry.